server/src/main/java/org/apache/cloudstack/agent/lb/IndirectAgentLBServiceImpl.java

```diff
@@ -136,17 +144,48 @@ public int compare(Long x, Long y) {
         }
         final List <Host> agentBasedHosts = new ArrayList<>();
         for (final Host host : allHosts) {
-            if (host == null || host.getResourceState() == ResourceState.Creating || host.getResourceState() == ResourceState.Error) {
-                continue;
+            conditionallyAddHost(agentBasedHosts, host);
+        }
+        return agentBasedHosts;
+    }
+
+    private void conditionallyAddHost(List<Host> agentBasedHosts, Host host) {
+        if (host == null) {
```

Comment:
We can do null checks and if host is removed or is some unsupported state like error etc.?